### ISSUE TITLE
all engine subtypes available, immediately

### ISSUE THREAD

**Issue by Schnobs** [KSP-RO/RealismOverhaul#1860]
Pulled RO-131 from ckan yesterday and added RP-0(dev) on top of it. Now all engines can be upgraded to any subtype as soon as the base model becomes available. Cost modifiers are not applied, though, so later types become quite expensive to unlock.

The RP-0 TREE-Engines.cfg looks fine, as does the MM ConfigCache. So I presume it's really a problem with RO or Realfuels or whatever mod has to apply these settings.

**Comment by raidernick** [comment]
It's definitely not RO. RO makes no changes to anything career related at all anywhere in the configs.

**Comment by Schnobs** [comment]
I know that the config comes from RP-0 and as best I can tell, the config is fine (diff between my last working config and the current one shows a few additions, but no sweeping change). But whatever mod is responsible for honoring the config no longer does so.

**Comment by jwvanderbeck** [comment]
RF handles the configs and exposes the upgrades.  Is that what you mean?

**Comment by Schnobs** [comment]
Trying to rephrase: 
On the WAC Corpral, my config (checked in MM ConfigCache) says the XASR upgrade should require _rocketryTesting_, and AJ10-27 _basicRocketryRP0_. But I can unlock / buy both configurations at start, before I've researched these nodes.

That's why I presume that the config itself is fine, and the actual problem is that the configured values are not being used.

I've fast forwarded a bit and seen the same issue with all other engines I tried.
